projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/templates/load/dimension/scd1/02-handle-quality-checks.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import re
+
+from vdk.api.job_input import IJobInput
+
+SQL_FILES_FOLDER = (
+    os.path.dirname(os.path.abspath(__file__)) + "/02-requisite-sql-scripts"
+)
+
+
+def run(job_input: IJobInput):
+    job_arguments = job_input.get_arguments()
+
+    check = job_arguments.get("check")
+    target_schema = job_arguments.get("target_schema")
+    target_table = job_arguments.get("target_table")
+    staging_schema = job_arguments.get("staging_schema")
+    insert_query = get_query("02-insert-into-target.sql")
+
+    if check:
+        if not staging_schema:
+            raise Exception(
+                "No staging_schema specified to execute the defined data checks against."
+            )
+
+        staging_table = f"{staging_schema}.{target_table}"
+        align_stg_table_with_target(
+            f"{target_schema}.{target_table}", staging_table, job_input
+        )
+
+        insert_into_staging = insert_query.format(
+            target_schema=staging_schema, target_table=target_table
+        )
+        job_input.execute_statement(insert_into_staging)
+
+        if check(staging_table):
+            insert_into_target = insert_query.format(
```

Comment:
Copying from staging table to target you probably do not want to reuse the same query. 

It works for this template because it's insert overwrite, but won't for the more complex templates. 

And we want to keep the code as for the next template we'd likely want to abstract teh common functionalities and reuse them.

Reply:
I will have to think of a more generic way to transfer the data in order to fit all templates needs.

Reply:
Add a TODO here if you are not ready to solve it now. It's easier to figure out what is "common" after a second instance of something.

Reply:
Okay lets proceed without that for now. 
We've discussed internally a way to introduce such generic functionality, but it will take time to be implemented, so we prefer to stick with what we have and introduce it on the next template modification.